### PR TITLE
parsing all files instead of only predefined filenames

### PR DESCRIPTION
**Pull request by yormy**
Recoded in a way that all tests remain functioning, non breaking changes.
Added a new setting so that the parsing of all files can be turned on when required

### CODE REVIEW

File: config/log-viewer.php [hunk 1]
```diff
@@ -22,6 +22,9 @@
         'extension' => Filesystem::PATTERN_EXTENSION, // '.log'
     ],
 
+    // parse all the files in the log path, regardless of their naming
+    'parse_all_files_in_log_path' => false,
```
Comment (arcanedev-maroc) [inline config/log-viewer.php]: Can you change the key to `parse-all-files-in-log-path` ?

File: src/Entities/LogCollection.php [hunk 2]
```diff
@@ -71,8 +71,14 @@ public function setFilesystem(FilesystemContract $filesystem)
      */
     private function load()
     {
-        foreach($this->filesystem->dates(true) as $date => $path) {
-            $this->put($date, Log::make($date, $path, $this->filesystem->read($date)));
+        if (config('log-viewer.parse_all_files_in_log_path')) {
```
Comment (arcanedev-maroc) [inline src/Entities/LogCollection.php]: Same here => `parse-all-files-in-log-path`

File: src/Http/Controllers/LogViewerController.php [hunk 3]
```diff
@@ -104,6 +104,10 @@ public function show($date, Request $request)
         $levels  = $this->logViewer->levelsNames();
         $entries = $log->entries($level)->paginate($this->perPage);
 
+        if (config('log-viewer.parse_all_files_in_log_path')) {
```
Comment (arcanedev-maroc) [inline src/Http/Controllers/LogViewerController.php]: Same here => `parse-all-files-in-log-path`

File: src/Http/Controllers/LogViewerController.php [hunk 4]
```diff
@@ -163,6 +167,10 @@ public function search($date, $level = 'all', Request $request)
      */
     public function download($date)
     {
+        if (config('log-viewer.parse_all_files_in_log_path')) {
```
Comment (arcanedev-maroc) [inline src/Http/Controllers/LogViewerController.php]: Same here => `parse-all-files-in-log-path`

File: src/Http/Controllers/LogViewerController.php [hunk 5]
```diff
@@ -178,6 +186,12 @@ public function delete(Request $request)
         abort_unless($request->ajax(), 405, 'Method Not Allowed');
 
         $date = $request->get('date');
+        if (config('log-viewer.parse_all_files_in_log_path')) {
```
Comment (arcanedev-maroc) [inline src/Http/Controllers/LogViewerController.php]: Same here => `parse-all-files-in-log-path`

File: src/Utilities/Filesystem.php [hunk 6]
```diff
@@ -106,6 +106,9 @@ public function setPath($storagePath)
      */
     public function getPattern()
     {
+        if (config('log-viewer.parse_all_files_in_log_path')) {
```
Comment (arcanedev-maroc) [inline src/Utilities/Filesystem.php]: Same here => `parse-all-files-in-log-path`

File: src/Utilities/Filesystem.php [hunk 7]
```diff
@@ -251,6 +261,9 @@ public function read($date)
     public function delete($date)
     {
         $path = $this->getLogPath($date);
+        if (config('log-viewer.parse_all_files_in_log_path')) {
```
Comment (arcanedev-maroc) [inline src/Utilities/Filesystem.php]: Same here => `parse-all-files-in-log-path`

File: src/Utilities/Filesystem.php [hunk 8]
```diff
@@ -319,6 +332,10 @@ private function getLogPath($date)
     {
         $path = $this->storagePath.DS.$this->prefixPattern.$date.$this->extension;
 
+        if (config('log-viewer.parse_all_files_in_log_path')) {
```
Comment (arcanedev-maroc) [inline src/Utilities/Filesystem.php]: Same here => `parse-all-files-in-log-path`

File: src/Utilities/LogMenu.php [hunk 9]
```diff
@@ -101,8 +101,15 @@ public function make(Log $log, $trans = true)
         $route = $this->config('menu.filter-route');
 
         foreach($log->tree($trans) as $level => $item) {
+
+            if (config('log-viewer.parse_all_files_in_log_path')) {
```
Comment (arcanedev-maroc) [inline src/Utilities/LogMenu.php]: Same here => `parse-all-files-in-log-path`

File: config/log-viewer.php (hunk 1 above)
Comment (yormy) [reply config/log-viewer.php]: Sure I can change the key. But about the test I do not know how to do that yet.
Flipping this switch will definately break existing tests. The only thing I can think of now is to include this feature switch also in ...all... testcases that use hardcoded naming
So I need some help how to sturcture coding these tests

Comment (arcanedev-maroc) [reply config/log-viewer.php]: You need to add new tests like:

```php
/** @test **/
public function it_can_load_all_log_files() 
{
    // Assert before switching

    $this->app['config']->set('log-viewer.parse-all-files-in-log-path', true);

    // Assert after switching
}
```

You can also create a new fixture folder that contains log files for your tests and switch it with `$this->logViewer->setPath(storage_path('my-path-logs'));`

Path: https://github.com/ARCANEDEV/LogViewer/tree/master/tests/fixtures
Example: https://github.com/ARCANEDEV/LogViewer/blob/master/tests/LogViewerTest.php#L312

**Comment by yormy** [comment]
Thanks, I created new tests and changed the config setting

**Comment by arcanedev-maroc** [comment]
Thanks a lot for your work @yormy.

There are some changes/cleaning to do, but i can take it from here.

Thanks again :+1:

**Pull request merged**